frontend/src/types/schemaSystem.ts

```diff
@@ -0,0 +1,351 @@
+import { SchemaReviewPolicyId } from "./id";
+import { Principal } from "./principal";
+
+export enum RuleLevel {
+  DISABLED = "DISABLED",
+  ERROR = "ERROR",
+  WARNING = "WARNING",
+}
+
+export const LEVEL_LIST = [
+  RuleLevel.ERROR,
+  RuleLevel.WARNING,
+  RuleLevel.DISABLED,
+];
+
+export enum PayloadType {
+  STRING = "STRING",
+  STRING_ARRAY = "STRING[]",
+  TEMPLATE = "TEMPLATE",
+}
+
+interface StringPayload {
+  type: PayloadType.STRING;
+  default: string;
+  value?: string;
+}
+
+interface StringArrayPayload {
+  type: PayloadType.STRING_ARRAY;
+  default: string[];
+  value?: string[];
+}
+
+interface TemplatePayload {
+  type: PayloadType.TEMPLATE;
+  default: string;
+  templateList: { id: string; description?: string }[];
+  value?: string;
+}
+
+interface NamingFormatRulePayload {
+  format: StringPayload | TemplatePayload;
+}
+
+interface RequiredColumnsRulePayload {
+  list: StringArrayPayload;
+}
+
+export type RulePayload =
+  | NamingFormatRulePayload
+  | RequiredColumnsRulePayload
+  | undefined;
+
+export type SchemaRuleEngineType = "MYSQL" | "COMMON";
+
+export type CategoryType = "ENGINE" | "NAMING" | "QUERY" | "TABLE" | "COLUMN";
+
+export interface SchemaRule {
+  id: string;
+  category: CategoryType;
+  engine: SchemaRuleEngineType;
+  description: string;
+  payload?: RulePayload;
+}
+
+export interface SelectedRule extends SchemaRule {
+  level: RuleLevel;
+}
+
+// rule payload
+interface NamingFormatPolicyPayload {
+  format: string;
+}
+
+interface RequiredColumnsPolicyPayload {
+  list: string[];
+}
+
+type PolicyPayload =
+  | NamingFormatPolicyPayload
+  | RequiredColumnsPolicyPayload
+  | undefined;
+
+interface DatabaseSchemaRule {
```

Comment:
My understanding is we need to have 2 rule type. One is the SchemaRule defined above or the SchemaRuleTemplate, and the other is the SchemaRule at runtime with levels and payload. But here we have 3 types: SchemaRule, SelectedRule and DatabaseSchemaRule.

Also the DatabaseSchemaRule contains a payload type PolicyPayload, which is a bit confusing since a rule should not contain a policy payload.

Reply:
Sure, SchemaRule and SelectedRule should combine into single interface.